handler/src/main/java/io/netty/handler/ssl/PemPrivateKey.java

```diff
@@ -60,7 +60,12 @@ static PemEncoded toPEM(ByteBufAllocator allocator, boolean useDirect, PrivateKe
             return ((PemEncoded) key).retain();
         }
 
-        ByteBuf encoded = Unpooled.wrappedBuffer(key.getEncoded());
+        byte[] bytes = key.getEncoded();
+        if (bytes == null) {
+            throw new IllegalArgumentException(key + " does not support encoding");
```

Comment:
`s/key/key.getClass().getName()/`?

Reply:
Right now it looks like it would give the inherited Object.toString() here for the most part. And we would hope it doesn't print out the private key. It's really the class name that is most important, so the swap cleans up the message.

Reply:
Good point... let me change it

Reply:
@ejona86 done